### PR TITLE
fleet mananager init

### PR DESCRIPTION
**Pull request by LiangquanLi930**
[OCPQE-13913](https://issues.redhat.com/browse/OCPQE-13913)

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by yunjiang29** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by yasun1** [comment]
I assume you are trying to provision the MC and SC clusters. The MC and SC clusters are the OSD CCS clusters but not the rosa cluster. You should use https://github.com/openshift/release/tree/master/ci-operator/step-registry/osd-ccs to prepare the cluster.

**Comment by LiangquanLi930** [comment]
> [osd-ccs](https://github.com/openshift/release/tree/master/ci-operator/step-registry/osd-ccs) to pr

So, we need to mv `rosa` to `osd-ccs`, rigrht?

**Comment by LiangquanLi930** [comment]
> @LiangquanLi930 I assume you are trying to provision the MC and SC clusters. The MC and SC clusters are the OSD CCS clusters but not the rosa cluster. You should use https://github.com/openshift/release/tree/master/ci-operator/step-registry/osd-ccs to prepare the cluster.

So, we need to mv rosa to osd-ccs, rigrht?

**Comment by yasun1** [comment]
@LiangquanLi930  I realized that the MC and the SC clusters are provisioned/deprovisioned via the OFM endpoints. It is different from the rosa hypershift clusters. You need to have a new folder to include the steps.

File: ci-operator/step-registry/osd-ccs/fleet-manager/check/osd-ccs-fleet-manager-check-ref.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  resources:
+    requests:
+      cpu: 100m
+      memory: 300Mi
+  env:
+  - name: ROSA_LOGIN_ENV
+    default: "staging"
+    documentation: The environment for rosa login. The supported values are [production, staging].
+  documentation: |-
+    xxxxxxxxxxx
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/check/osd-ccs-fleet-manager-check-ref.yaml:18]: What is the `check` for?

File: ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+workflow:
+  as: osd-ccs-fleet-manager
+  steps:
+    pre:
+    - chain: osd-ccs-fleet-manager-provision
+    test:
+    - ref: osd-ccs-fleet-manager-check
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml:7]: The workflow should be used to providing the clusters, but not doing testing.

File: ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+workflow:
+  as: osd-ccs-fleet-manager
+  steps:
+    pre:
+    - chain: osd-ccs-fleet-manager-provision
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml:5]: The workflow is the lifecycle on provisioning/deprovisioning the mc/sc clusters via the fleet manager API, but not provisioning/deprovisioning the fleet manager.

File: ci-operator/step-registry/osd-ccs/fleet-manager/provision/sc/osd-ccs-fleet-manager-provision-sc-commands.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token fleetmanager_TOKEN!"
+  exit 1
+fi
+
+#rosa whoami
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/provision/sc/osd-ccs-fleet-manager-provision-sc-commands.sh:52]: It should use ocm.

**Comment by yasun1** [comment]
@LiangquanLi930 @YuLi517  On my side, I'd like to have a design doc rather than deliver a draft structure into release. We'd better deliver the completed steps. The `release` repository is not our private repo, many teams work on this, and many eyes are on this, it is a good showing to the others that we input an uncompleted work.

**Comment by LiangquanLi930** [comment]
> @LiangquanLi930 @YuLi517 On my side, I'd like to have a design doc rather than deliver a draft structure into release. We'd better deliver the completed steps. The `release` repository is not our private repo, many teams work on this, and many eyes are on this, it is a good showing to the others that we input an uncompleted work.

Hi, I think there's nothing wrong with that, Please look at this, https://github.com/openshift/release/pull/32839/files

File: ci-operator/step-registry/osd-ccs/fleet-manager/deprovision/osd-ccs-fleet-manager-deprovision-chain.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+chain:
+  as: osd-ccs-fleet-manager-deprovision
+  steps:
+  - ref: osd-ccs-fleet-manager-deprovision-sc
```
[the remark this answers is not in the document]
Comment (LiangquanLi930) [reply ci-operator/step-registry/osd-ccs/fleet-manager/deprovision/osd-ccs-fleet-manager-deprovision-chain.yaml:4]: This is just a design, a better expansion in the future, we will also join the new Step in the future

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-rosa-fleet-manager-p2-f21

File: ci-operator/step-registry/osd-ccs/fleet-manager/sc/deprovision/osd-ccs-fleet-manager-sc-deprovision-ref.yaml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    requests:
+      cpu: 100m
+      memory: 300Mi
+  env:
+  - name: OCM_LOGIN_ENV
+    default: "staging"
+    documentation: The environment for ocm login. The supported values are [production, staging].
+  - name: CLOUD_PROVIDER_REGION
+    default: ""
+    documentation: Set a region for the ocm cluster creation. If not specified, a cluster_profile appropriate setting will be used.
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/sc/deprovision/osd-ccs-fleet-manager-sc-deprovision-ref.yaml:19]: Do not need the region in the devprovision step.

File: ci-operator/step-registry/osd-ccs/fleet-manager/sc/provision/osd-ccs-fleet-manager-sc-provision-ref.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+      memory: 300Mi
+  env:
+  - name: OCM_LOGIN_ENV
+    default: "staging"
+    documentation: The environment for ocm login. The supported values are [production, staging].
+  - name: CLOUD_PROVIDER_REGION
+    default: ""
+    documentation: Set a region for the ocm cluster creation. If not specified, a cluster_profile appropriate setting will be used.
+  documentation: |-
+    Create sc using ocm cli.
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/sc/provision/osd-ccs-fleet-manager-sc-provision-ref.yaml:21]: Do not use the abbreviation word `sc` in the description. And the description should be like 'Using the OSD Fleet Manager api endpoint to provision the service cluster, and a default management cluster will be provisioned together. '

File: ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+workflow:
+  as: osd-ccs-fleet-manager
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml:2]: It should be put under the sc folder, and the workflow is `osd-ccs-fleet-manager-sc`

File: ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+  steps:
+    pre:
+    - ref: osd-ccs-fleet-manager-sc-provision
+    post:
+    - ref: osd-ccs-fleet-manager-sc-deprovision
+  documentation: |-
+    osd ccs fleet manager
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/osd-ccs-fleet-manager-workflow.yaml:9]: The description should be `The workflow installs a service cluster and a management cluster with the specific configuration. After finishing testing, both of the two clusters will be deprovsioned.`

File: ci-operator/step-registry/osd-ccs/fleet-manager/sc/deprovision/osd-ccs-fleet-manager-sc-deprovision-ref.yaml [hunk 10]
```diff
@@ -0,0 +1,21 @@
+ref:
+  as: osd-ccs-fleet-manager-sc-deprovision
+  from_image:
+    namespace: ci
+    name: cli-ocm
+    tag: latest
+  grace_period: 3h10m
```
Comment (yasun1) [inline ci-operator/step-registry/osd-ccs/fleet-manager/sc/deprovision/osd-ccs-fleet-manager-sc-deprovision-ref.yaml:7]: grace_period is the time used to shut down the pod gracefully after the pod finishes its work. `3h10m` is a too big number, we should not wait for so long times,`10m` is enough.

Comment (LiangquanLi930) [reply ci-operator/step-registry/osd-ccs/fleet-manager/sc/deprovision/osd-ccs-fleet-manager-sc-deprovision-ref.yaml:7]: thanks a lot! learning day!

**Pull request merged**